### PR TITLE
refactor(mission_planner): delete default values

### PR DESCRIPTION
**Pull request by yamazakiTasuku**
## Description
Removed default values defined in declare_parameter function.
[mission_planner_delete_param.webm](https://user-images.githubusercontent.com/100691117/226554272-90b2e7d7-88c7-4d98-80b3-31e1a2183107.webm)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/mission_planner/src/lanelet2_plugins/default_planner.cpp [hunk 1]
```diff
@@ -145,8 +145,8 @@ void DefaultPlanner::initialize_common(rclcpp::Node * node)
     node_->create_publisher<MarkerArray>("debug/goal_footprint", durable_qos);
 
   vehicle_info_ = vehicle_info_util::VehicleInfoUtil(*node_).getVehicleInfo();
-  param_.goal_angle_threshold_deg = node_->declare_parameter("goal_angle_threshold_deg", 45.0);
-  param_.enable_correct_goal_pose = node_->declare_parameter("enable_correct_goal_pose", false);
+  param_.goal_angle_threshold_deg = node_->declare_parameter("goal_angle_threshold_deg");
+  param_.enable_correct_goal_pose = node_->declare_parameter("enable_correct_goal_pose");
```
Comment (isamu-takagi) [inline planning/mission_planner/src/lanelet2_plugins/default_planner.cpp:149]: A template type is necessary.

Comment (satoshi-ota) [reply planning/mission_planner/src/lanelet2_plugins/default_planner.cpp:149]: @isamu-takagi Thanks for your comment.

@yamazakiTasuku Could you fix it?

**Comment by satoshi-ota** [comment]
@yamazakiTasuku Thanks for your work. Please let me comment.

Please don't check following checkbox before the review was **NOT** finished :pray: 

image

File: planning/mission_planner/src/lanelet2_plugins/default_planner.cpp (hunk 1 above)
Comment (satoshi-ota) [reply planning/mission_planner/src/lanelet2_plugins/default_planner.cpp]: I fixed in https://github.com/autowarefoundation/autoware.universe/pull/3127/commits/2e7e30ae007d36da3b199c15b88a92673058e8a6.

**Review by satoshi-ota — APPROVED** [review]
These default values are defined here. LGTM.

https://github.com/autowarefoundation/autoware_launch/blob/133a3d94b87ff7278f1b0585747b762609c645e1/autoware_launch/config/planning/mission_planning/mission_planner/mission_planner.param.yaml#L7-L8

**Comment by isamu-takagi** [comment]
@yamazakiTasuku 
Could you apply this change to `static_centerline_optimizer` to pass CI? This package depends on `mission_planner` [here](https://github.com/autowarefoundation/autoware.universe/blob/main/planning/static_centerline_optimizer/src/static_centerline_optimizer_node.cpp#L307-L318).

**Comment by satoshi-ota** [comment]
@yamazakiTasuku @isamu-takagi I take over this PR.

**Review by satoshi-ota — CHANGES_REQUESTED** [review]
Add mission_planner.param.yaml

https://github.com/autowarefoundation/autoware.universe/blob/a3a5b19d9ac49f611b3552a28a807a63a0e11beb/planning/static_centerline_optimizer/test/test_static_centerline_optimizer.test.py#L34-L38

**Review by satoshi-ota — APPROVED** [review]

**Comment by satoshi-ota** [comment]
@kosuke55 @isamu-takagi I would like you to check this PR as code owner.

File: planning/static_centerline_optimizer/test/test_static_centerline_optimizer.test.py [hunk 2]
```diff
@@ -43,6 +43,10 @@ def generate_test_description():
             {"lanelet2_output_file_path": "/tmp/lanelet2_map.osm"},
             {"start_lanelet_id": 215},
             {"end_lanelet_id": 216},
+            os.path.join(
+                get_package_share_directory("mission_planner"),
+                "config/mission_planner.param.yaml",
```
Comment (kosuke55) [inline planning/static_centerline_optimizer/test/test_static_centerline_optimizer.test.py:48]:
```suggestion
                "config",
                mission_planner.param.yaml",
```
better for multi-platform

**Review by kosuke55 — APPROVED** [review]

**Review by isamu-takagi — APPROVED** [review]

**Pull request merged**